train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -46,4 +47,9 @@ public void setJoystickPosition(int joystickPosition) {
 		this.step = joystickPosition;		
 	}
 
+
+	public TrainControllerImpl(){
+		Executors.newSingleThreadScheduledExecutor().scheduleAtFixedRate(this::followSpeed,0,200,TimeUnit.MILLISECONDS);
+	}
+
```

Comment:
Thread safety not ensured in code, nor explained in comment/documentation why this can not lead to actual hazard/failure! This will be calling a mutating method from another thread, potentially leading to a race condition.